### PR TITLE
MAGECLOUD-3510: Fix an Issue with db-dump Command Related to DEFINERS

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
Add functionality for removing definers during ece-tool `db-dump` command.
Added option `--remove-definers` `-d` to remove definers from the dumped file.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3510

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-13

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by bbatsche — DISMISSED** [review]

File: src/DB/DumpGenerator.php [hunk 1]
```diff
@@ -84,7 +82,7 @@ public function __construct(
      *
      * {@inheritdoc}
      */
-    public function execute()
+    public function create(bool $keepDefiners = false)
```
Comment (shiftedreality) [inline src/DB/DumpGenerator.php]: Let's not use optional values as it adds complexity

File: src/Command/DbDump.php [hunk 2]
```diff
@@ -49,6 +52,13 @@ protected function configure()
         $this->setName(self::NAME)
             ->setDescription('Creates backup of database');
 
+        $this->addOption(
+            self::OPTION_KEEP_DEFINERS,
+            'k',
+            InputOption::VALUE_NONE,
```
Comment (shiftedreality) [inline src/Command/DbDump.php:58]: Let's use `false` as default value

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/Command/DbDump.php (hunk 2 above)
Comment (oshmyheliuk) [reply src/Command/DbDump.php:58]: For `InputOption::VALUE_NONE` false is the default state, don't see a point to add this.

File: src/DB/DumpGenerator.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/DB/DumpGenerator.php]: Please give your suggestion on how to avoid it in this particular case

Comment (oshmyheliuk) [reply src/DB/DumpGenerator.php]: done

**Review by shiftedreality — DISMISSED** [review]

**Review by arhiopterecs — DISMISSED** [review]

**Review by shiftedreality — APPROVED** [review]

**Review by arhiopterecs — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**